src/charmed_kubeflow_chisme/components/pebble_component.py

```diff
@@ -28,7 +28,32 @@ def __init__(
         group: Optional[str] = None,
         permissions: Optional[str] = None,
     ):
-        """A ContainerFileTemplate that lazily generates its template from a path or string."""
+        """A lazy file template renderer for use in pushing files to a Pebble container.
+
+        This generates a file from a template provided as a string or a file path.  The file will
+        be rendered with a context provided as a dict.
+
+        source_template_path, source_template, and context can also be provided as callables that
+        return the appropriate value.  This allows for lazy evaluation of these values, which can
+        be useful for using in charm Components.
+
+        Args:
+            destination_path: The path to the file in the container.
+            source_template_path: The path to the source template file.
+                                  Can be provided as a Path, str, or a function that returns
+                                  either.
+                                  Only one of source_template_path or source_template can be set.
+            source_template: The source template string.  Only one of source_template_path or this
+                             can be set.
+                             Can be provided as a str or a function that returns a str.
```

Comment:
```suggestion
            source_template: The source template string.
                             Can be provided as a str or a function that returns a str.
                             Only one of source_template_path or this can be set.
```